libraries/adaptive-expressions/src/expressionEvaluator.ts

```diff
@@ -5,21 +5,32 @@
  * Copyright (c) Microsoft Corporation. All rights reserved.
  * Licensed under the MIT License.
  */
-import { Expression, ReturnType } from './expression';
+import { Expression } from './expression';
 import { MemoryInterface } from './memory';
 import { Options } from './options';
+import { ReturnType } from './returnType';
 
 /**
  * Delegate for doing static validation on an expression.
  * Validators can and should throw exceptions if the expression is not valid.
  */
 export type ValidateExpressionDelegate = (expression: Expression) => any;
 
+/**
+ * Value result with error.
+ */
+// eslint-disable-next-line @typescript-eslint/prefer-interface
+export type ValueWithError = 
```

Comment:
>type [](start = 7, length = 4)

Couldn't this be an interface?

Reply:
Change the type to the interface is a breaking change. For someone using this package would broken.
For example:
```
interface ValueWithError  {value: any; error: string|undefined}
public foo(): ValueWithError  {xxx}
```

Someone using the foo() in other libraries with old style and without interface
```
bar():{value: any; error: string|undefined} // would build failed {
 return foo();
}
```